services/src/main/java/com/chatbot/services/DialogflowServices/DialogflowConversation.java

```diff
@@ -0,0 +1,123 @@
+import com.google.cloud.dialogflow.v2.QueryParameters;
+import com.google.cloud.dialogflow.v2.QueryResult;
+import com.google.cloud.dialogflow.v2.SessionName;
+import com.google.cloud.dialogflow.v2.SessionsClient;
+import com.google.cloud.dialogflow.v2.TextInput;
+import com.google.protobuf.Struct;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.cloud.dialogflow.v2.EventInput;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DialogflowConversation {
+
+  private static final Logger logger = LoggerFactory.getLogger(DialogflowConversation.class);
+  private static String projectID;
+  private final String langCode;
+  private final String sessionID;
+
+  DialogflowConversation(final String projectIDToSet, final String langCodeToSet,
+      final String sessionIDToSet) {
+    projectID = projectIDToSet;
+    langCode = langCodeToSet;
+    sessionID = sessionIDToSet;
+  }
+
+  public DialogflowConversation(final String projectIDToSet, final String sessionIDToSet) {
+    this(projectIDToSet, "en", sessionIDToSet);
+  }
+
+  // function to get the response for a user message from dialogflow
+  public QueryResult sendMessage(final String message, final Struct payload) throws Exception {
+    try (SessionsClient sessionsClient = SessionsClient.create()) {
+      final SessionName session = SessionName.of(projectID, sessionID);
+      final TextInput.Builder textInput = TextInput.newBuilder().setText(message)
+          .setLanguageCode(langCode);
+      final QueryInput queryInput = QueryInput.newBuilder().setText(textInput).build();
+      final QueryParameters queryParameters = QueryParameters.newBuilder().setPayload(payload)
+          .build();
+      final DetectIntentRequest detectIntentRequest =
+          DetectIntentRequest.newBuilder().setSession(session.toString()).setQueryInput(queryInput)
+          .setQueryParams(queryParameters).build();
+      final DetectIntentResponse response = sessionsClient.detectIntent(detectIntentRequest);
+      return response.getQueryResult();
+    }
+  }
+
+  // function to get the response for an event from dialogflow
+  public QueryResult triggerEvent(final String event, final Struct parameters, final Struct payload)
+      throws Exception {
+    try (SessionsClient sessionsClient = SessionsClient.create()) {
+      final SessionName session = SessionName.of(projectID, sessionID);
+      final EventInput.Builder eventInput = EventInput.newBuilder()
+          .setName(event)
+          .setParameters(parameters)
+          .setLanguageCode(langCode);
+      final QueryInput queryInput = QueryInput.newBuilder().setEvent(eventInput).build();
+      final QueryParameters queryParameters = QueryParameters.newBuilder()
+          .setPayload(payload)
+          .build();
+      final DetectIntentRequest detectIntentRequest = DetectIntentRequest.newBuilder()
+          .setSession(session.toString())
```

Comment:
Extract it out in a private function to avoid repetition as this is also used in sendMessage

Reply:
updated